compute_endpoint/globus_compute_endpoint/cli.py

```diff
@@ -743,40 +763,71 @@ def list_endpoints():
 
 
 @app.command("delete")
-@name_or_uuid_arg
+@name_or_uuid_arg_non_local
 @click.option(
     "--force",
     default=False,
     is_flag=True,
-    help="Deletes the local endpoint even if the web service returns an error.",
+    help=(
+        "Attempts to delete the endpoint both locally and remotely even if it "
+        "is currently running, does not exist locally, has an invalid config, "
+        "or an error is encountered trying to connect to the Globus Compute web "
+        "services."
+    ),
 )
 @click.option(
-    "--yes", default=False, is_flag=True, help="Do not ask for confirmation to delete."
+    "--yes",
+    default=False,
+    is_flag=True,
+    help="Do not ask for confirmation to delete endpoints.",
 )
 @common_options
 @handle_auth_errors
-def delete_endpoint(*, ep_dir: pathlib.Path, force: bool, yes: bool):
+def delete_endpoint(
+    *,
+    ep_dir: pathlib.Path | None,
+    ep_uuid: str | None = None,
+    force: bool,
+    yes: bool,
+):
     """Deletes an endpoint and its config."""
 
-    ep_conf = None
-    try:
-        ep_conf = get_config(ep_dir)
-    except Exception as e:
-        print(f"({type(e).__name__}) {e}\n")
-        if not yes:
-            yes = click.confirm(
-                f"Failed to read configuration from {ep_dir}/\n"
-                f"  Are you sure you want to delete endpoint <{ep_dir.name}>?",
-                abort=True,
-            )
+    ep_info = f"< {ep_dir.name} >" if ep_dir else ""
+    if ep_uuid:
+        ep_info += f" ( {ep_uuid} )"
 
     if not yes:
-        yes = click.confirm(
-            f"Are you sure you want to delete endpoint <{ep_dir.name}>?",
+        click.confirm(
+            f"Are you sure you want to delete endpoint {ep_info}?",
             abort=True,
         )
 
-    Endpoint.delete_endpoint(ep_dir, ep_conf, force=force)
+    ep_conf = None
+    error_msg = None
+    try:
+        if ep_dir:
+            ep_conf = get_config(ep_dir)
+        else:
+            error_msg = f"No local endpoint configuration available for {ep_uuid}"
+    except Exception as e:
+        if isinstance(e.__cause__, FileNotFoundError):
+            error_msg = f"{ep_dir}/ does not exist on the file system"
+        else:
+            # Endpoint directory exists locally but error loading the config
+            error_msg = f"Failed to read configuration from {ep_dir}/ "
```

Comment:
```suggestion
    except FileNotFoundError:
        error_msg = f"{ep_dir}/ does not exist on the file system"
    except Exception:
        # Endpoint directory exists locally but error loading the config
        error_msg = f"Failed to read configuration from {ep_dir}/ "
```

Reply:
splitting the two exceptions is no longer relevant - as discussed, the FileNotFoundError (checking for endpoint dir doesn't exist) was moved to the decorator so the main delete_endpoint() body shouldn't encounter it.